lib/BackgroundJob/RetentionJob.php

```diff
@@ -219,7 +234,14 @@ private function checkFileId(int $fileId): Node {
 			throw new NotFoundException('No node for file ' . $fileId . ' and user ' . $userId);
 		}
 
-		return array_shift($nodes);
+		foreach ($nodes as $node) {
+			if ($node->getPermissions() & \OCP\Constants::PERMISSION_DELETE) {
+				return $node;
+			}
+			$this->logger->debug('Mount point has access to node ' . $node->getId() . ' but permissions are ' . $node->getPermissions());
```

Comment:
I would add the mount point information too to this message for the sake of completeness.